R/model_performance.R

```diff
@@ -139,3 +147,57 @@ model_performance_accuracy <- function(tp, fp, tn, fn) {
   (tp + tn)/(tp + fp + tn + fn)
 }
 
+model_performance_macro_f1 <- function(predicted, observed){
+  predicted_vectorized <- turn_porbs_into_vector(predicted)
+  confusion_matrixes <- calculate_confusion_matrixes(predicted_vectorized, observed)
+  f1_scores <- sapply(confusion_matrixes, function(x){
```

Comment:
lets use the style guid in which there is space between `)` and `{`
as in other functions

Reply:
Done